runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -2610,14 +2610,30 @@ class INTERPRETER_CLASS
 		_sp += (slotCount - 1);
 	}
 
-	/* java.lang.Thread: public static native Thread _currentThread(); */
+	/* java.lang.Thread: public static native Thread currentThread(); */
 	VMINLINE VM_BytecodeAction
 	inlThreadCurrentThread(REGISTER_ARGS_LIST)
 	{
 		returnObjectFromINL(REGISTER_ARGS, _currentThread->threadObject, 0);
 		return EXECUTE_BYTECODE;
 	}
 
+#if JAVA_SPEC_VERSION >= 19
+	/* java.lang.Thread: native void setCurrentThread(Thread thread); */
+	VMINLINE VM_BytecodeAction
+	inlThreadSetCurrentThread(REGISTER_ARGS_LIST)
+	{
+		j9object_t thread = ((j9object_t*)_sp)[0];
+		j9object_t receiverObject = ((j9object_t*)_sp)[1];
+		J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(_currentThread, receiverObject);
+		/* This is a package private method, currently the receiver object is Thread.currentCarrierThread()
```

Comment:
Since changes are needed elsewhere, please put the close comment on its own line and copy this comment to the fast version.

Reply:
Updated.